cmd/stat.go

```diff
@@ -215,5 +231,162 @@ func statURL(ctx context.Context, targetURL, versionID string, isIncomplete, isR
 		stats = append(stats, stat)
 	}
 
-	return stats, probe.NewError(cErr)
+	return stats, bucketStats, probe.NewError(cErr)
+}
+
+// BucketInfo holds info about a bucket
+type BucketInfo struct {
+	Versioning struct {
+		Status    string `json:"status"`
+		MFADelete string `json:"MFADelete"`
+	} `json:"Versioning,omitempty"`
+	Encryption struct {
+		Algorithm string `json:"algorithm,omitempty"`
+		KeyID     string `json:"keyId,omitempty"`
+	} `json:"Encryption,omitempty"`
+	Locking struct {
+		Enabled  string              `json:"enabled"`
+		Mode     minio.RetentionMode `json:"mode"`
+		Validity string              `json:"validity"`
+	} `json:"ObjectLock,omitempty"`
+	Replication struct {
+		Enabled bool               `json:"enabled"`
+		Config  replication.Config `json:"config,omitempty"`
+	} `json:"Replication"`
+	Policy struct {
+		Type string `json:"type"`
+		Text string `json:"policy"`
+	} `json:"Policy,omitempty"`
+	Location string            `json:"location"`
+	Tagging  map[string]string `json:"tagging,omitempty"`
+	ILM      struct {
+		Config *lifecycle.Configuration `json:"config"`
+	} `json:"ilm,omitempty"`
+	Notification struct {
+		Config notification.Configuration `json:"config,omitempty"`
+	} `json:"notification,omitempty"`
+}
+
+// Tags returns stringified tag list.
+func (i BucketInfo) Tags() string {
+	keys := []string{}
+	maxKeyLen := 0
+	for key := range i.Tagging {
+		keys = append(keys, key)
+		if len(key) > maxKeyLen {
+			maxKeyLen = len(key)
+		}
+	}
+	sort.Strings(keys)
+
+	maxKeyLen += 2 // add len(" :")
+	strs := []string{}
+	for _, key := range keys {
+		strs = append(
+			strs,
+			fmt.Sprintf("%v%*v %v", console.Colorize("Key", key), maxKeyLen-len(key), ":", console.Colorize("Value", i.Tagging[key])),
+		)
+	}
+
+	return strings.Join(strs, "\n\t ")
+}
+
+type bucketInfoMessage struct {
+	Op     string
+	Status string     `json:"status"`
+	URL    string     `json:"url"`
+	Info   BucketInfo `json:"info"`
+}
+
+func (v bucketInfoMessage) JSON() string {
+	v.Status = "success"
+	buf := &bytes.Buffer{}
+	enc := json.NewEncoder(buf)
+	enc.SetIndent("", " ")
+	// Disable escaping special chars to display XML tags correctly
+	enc.SetEscapeHTML(false)
+
+	fatalIf(probe.NewError(enc.Encode(v)), "Unable to marshal into JSON.")
+	return buf.String()
+
+}
+
+func (v bucketInfoMessage) String() string {
+	var b = &strings.Builder{}
```

Comment:
```
var b strings.Builder
```

Is the idiomatic style @poornas you need to allocate a strings builder. 

>  The zero value is ready to use. 

https://golang.org/pkg/strings/#Builder